### DISCUSSION TITLE
R - Geometric Distribution - Probability of a machine lasting X days

### DISCUSSION THREAD

**Discussion by Hertzspr** [teamalgoritma/community#38] [Q&A] [answered]
Suppose a new machine arrives in a factory. The machine has a 30% chance of breaking permanently. 

If the probability the machine breaks on 5th day or earlier is calculated with: 
`pgeom(4, .3)`

Why does the probability the machine is still working on 20th day is calculated with:
`1 - pgeom(19, .3)`

Why not calculated with:
`pgeom(20, .3)`

---

Kenapa **kemungkinan mesin rusak di hari ke 21** beda dengan **kemungkinan mesin masih bekerja di hari ke 20**?

**Comment by onlyphantom** [comment, marked as answer]
- `pgeom(0, 0.3)` = 0.3 (_a priori_)
- `pgeom(5, 0.3)`: Machine breaks down by 5th day  or earlier
- `pgeom(19, 0.3)`: Machine breaks down by 19th day  or earlier
- `pgeom(20, 0.3)`: Machine breaks down by 20th day or earlier

This also intuitively explains why `pgeom(20, 0.3)` is 0.9994415, a lot greater than `pgeom(4, 0.3)` which is 0.83193. 

Probabilities sum up to one. So if the question is "probability the machine is still working on 20th day", that is the same as:
-  `Total Probability  - probability of machine breaking down by 19th day or earlier`
-  `1 - p(break down by 19th day or earlier)`

Plug that into the earlier equation:
`1 - pgeom(19, .3)`. It's saying that with a machine so likely to break down by the 19th day or earlier (0.9992021) , the chance of it _not_ breaking down (and therefore, _survive to the next day_, day 20) is only 0.0007979227.

### Additional explanation

The `pgeom` function in your example above finds the probability of experiencing a certain amount of success or less before experiencing the first failure in a series of Bernoulli trials (geometric cumulative density function.). In this case, 'success' means survival of the machine for that one day (0.3). 

<img alt="pgeom" src="https://user-images.githubusercontent.com/16984453/161964078-128c731f-386d-4bb3-88e8-812efbe4cd25.jpg" width=200 />

Area under graph = total probability = 1
If `pgeom(20, x)` calculates the probability of survival in 20 days or less (20 success or less) before experience the first failure, then we draw a dotted line on the distribution (image above) and gray out the area on the left. Taking the right-hand side of the curve is just 1 minus the grayed out area.

**Reply by Hertzspr** [reply]
Thank you.
